accepted/2021/preview-features/preview-features.md

```diff
@@ -0,0 +1,418 @@
+isn't a supported part of .NET, therefore she should make sure to not integrate
+the library into the production instance of Fabrikam.
+
+## Requirements
+
+### Goals
+
+* Customers can discover preview features
+* Customers are made aware if a particular feature is in a preview state so that
+  they always know which part of the product is fully supported and which parts
+  aren't.
+* Customer have an easy way to audit if their applications or libraries are
+  using preview features.
+* Customers can't accidentally depend on preview features, either directly or
+  indirectly through some library.
+* We can deliver a reasonable experience even when a preview feature can't be
+  supported in all the workloads, for example, by failing with a sensible error
+  message.
+* We have a unified experience for enabling preview features, no matter whether
+  they are in a runtime, a compiler, a library, or span multiple parts of the
+  system.
+* We don't have to do unnatural acts to expose a preview feature, by, for
+  example, having to build a plug-in based system that allows swapping in and
+  out files.
+
+### Non-Goals
+
+* Allowing Microsoft to turn features on or off remotely, e.g. A/B testing
+* Designing a mechanism for shipping preview features as separate packages or
+  installers
+
+## Stakeholders and Reviewers
+
+* Runtime teams
+* C#, VB, F# compiler and IDE teams
+* Library teams
+* Project system team
+* SDK team
+* MSBuild team
+* NuGet team
+
+## Design
+
+### Property in project files
+
+```xml
+<Project Sdk="Microsoft.NET.Sdk">
+
+  <PropertyGroup>
+    <TargetFramework>net6.0</TargetFramework>
+    <EnablePreviewFeatures>True</EnablePreviewFeatures>
+  </PropertyGroup>
+
+</Project>
+```
+
+### Assembly info generation
+
+When we generate the `AssemblyInfo.cs`, we'll generate an assembly level
+attribute (the attribute and the meaning is discussed [later in this
+document](#api-analyzer)).
+
+For example, this project information:
+
+```xml
+<Project>
+
+  <PropertyGroup>
+    <TargetFramework>net6.0</TargetFramework>
+    <EnablePreviewFeatures>True</EnablePreviewFeatures>
+  </PropertyGroup>
+
+</Project>
+```
+
+Would result in this assembly-level attribute:
+
+```C#
+[assembly: RequiresPreviewFeatures]
+```
+
+This means you can look at a binary and tell whether it was built using preview
+features. This can be used for auditing and for detecting transitive
+dependencies ([discussed later](#api-analyzer)).
+
+### runtime.json
+
+Whether or not `EnablePreviewFeatures` was set to true for the application
+should be written into the `runtime.json` file such that the host/runtime can
+use it to make decisions. Alternatively, we could say this is encoded by the
+fact that the assembly the host considers the entry point has the
+`RequiresPreviewFeatures` assembly level attribute.
+
+### Compilation context
+
+When `EnablePreviewFeatures` is true, the `LangVersion` property should be set
```

Comment:
Does this mean that we can forever assume that if `EnablePreviewFeatures` is `true` then `LangVersion` was also `true`? Should everyone be able to infer that or is that an implementation detail at present?

Reply:
`EnablePreviewFeatures` should set `LangVersion` to `preview` unless the user explicitly set a value for `LangVersion`.